multicluster/config/samples/clusterset_init/multicluster_clusterset_template.yaml

```diff
@@ -20,9 +20,4 @@ metadata:
 spec:
   leaders:
     - clusterID: test-cluster-leader
```

Comment:
@luolanzone: is `leaders` needed for a leader cluster ClusterSet or it can be removed?

Reply:
It's required for leader cluster set validation. we probably can remove it in the future if we remove the step to comparing cluster ID with the ID in ClusterClaim.

Reply:
I feel such a check is not necessary. But probably let us keep the field for now.